src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,70 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
 
     public void getStatistic(String fromFileName, String toFileName) {
+        int[] data = readDataFromFile(fromFileName);
+        writeDataToFile(toFileName, data[0], data[1]);
+    }
+
+    private int[] readDataFromFile(String file) {
+        try {
+            BufferedReader readSupply = new BufferedReader(new FileReader(file));
+            int[] data = new int[2];
+            data[0] = getSupplyCount(readSupply);
+            BufferedReader readSales = new BufferedReader(new FileReader(file));
+            data[1] = getSellCount(readSales);
```

Comment:
you are iterating through the file twice. Add just one else to check whether it's a but or a supply and you can make it with once circle